[web] fix: allow custom default port

jaulz opened the pull request:
This fix will allow you to use a custom port for the webpack dev server which is especially handy if you need a specific port in development (e.g. because of CORS).

EvanBacon commented:
```js
const port = parseInt(process.env.PORT, 10) || 19006;
```
Seems more standard. Then we can support other tools like https://github.com/c9/core/blob/d8917f33fd7d32141d895fb7de530cfee2a416cf/configs/standalone.js#L15. 

Also not sure is `process.env.IP` or `process.env.HOST` is better for the host value.

EvanBacon commented on packages/xdl/src/Webpack.js:
```diff
@@ -18,8 +18,8 @@ import ip from './ip';
 
-const HOST = '0.0.0.0';
-const DEFAULT_PORT = 19006;
+const HOST = process.env.IP || '0.0.0.0';
```
If you add this functionality you'll need to account for the URL getter https://github.com/expo/expo-cli/blob/f5d61b6fcc87cadaefc578ee770185fcc69fabff/packages/xdl/src/Webpack.js#L117-L125. This may need to read from the running instance of Webpack to ensure the correct IP is being used. Currently it's hardcoded.

jaulz commented:
Good point! I added the PORT environment variable but still respect the port defined in the webpack dev  server config. In this case we will be flexible for any future requirements.

Furthermore I added the `PUBLIC_HOST` constant and adjusted the URL getter correspondingly.
